ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md

```diff
@@ -0,0 +1,193 @@
+---
+layout: news_post
+title: "Ruby 3.4.0 rc1 릴리스"
+author: "naruse"
+translator: "shia"
+date: 2024-12-12 00:00:00 +0000
+lang: ko
+---
+
+{% assign release = site.data.releases | where: "version", "3.4.0-rc1" | first %}
+Ruby {{ release.version }} 릴리스를 알리게 되어 기쁩니다.
+
+## Prism
+
+parse.y에서 Prism으로 기본 파서를 변경했습니다. [[Feature #20564]]
+
+## 모듈러 GC
+
+* 다른 가비지 컬렉터(GC) 구현을 모듈러 가비지 컬렉터 기능을 통해 동적으로 로드할 수 있습니다.
+  이 기능을 활성화하려면 Ruby 빌드 시에 `--with-modular-gc`를 설정하세요. GC 라이브러리는
+  환경 변수 `RUBY_GC_LIBRARY`를 사용하여 런타임에 로드할 수 있습니다.
+  [[Feature #20351]]
+
+* Ruby의 내장 가비지 컬렉터는 `gc/default/default.c`에 분리되어 있으며,
+  `gc/gc_impl.h`에 정의된 API를 사용하여 Ruby와 상호 작용합니다.
+  내장 가비지 컬렉터는 `make modular-gc MODULAR_GC=default`를 사용하여
+  라이브러리로서 빌드하고 환경 변수 `RUBY_GC_LIBRARY=default`를
+  사용하여 활성화할 수 있습니다. [[Feature #20470]]
+
+* [MMTk](https://www.mmtk.io/)를 기반으로 한 실험적인 GC 라이브러리가 제공됩니다.
+  이 GC 라이브러리는 `make modular-gc MODULAR_GC=mmtk`를 사용하여 빌드하고
+  환경 변수 `RUBY_GC_LIBRARY=mmtk`를 사용하여 활성화할 수 있습니다.
+  이는 빌드 머신에 Rust 도구 체인이 필요합니다. [[Feature #20860]]
+
+
+## 언어 변경
+
+* 파일에 `frozen_string_literal` 주석이 없을 때, 문자열 리터럴이 변경되면
+  폐기 예정 경고를 출력합니다.
+  이 경고는 `-W:deprecated`나 `Warning[:deprecated] = true` 설정을 통해 활성화할 수 있습니다.
+  이 변경을 무효화하고 싶다면 Ruby를 실행할 때 `--disable-frozen-string-literal` 커맨드라인 인수를
+  사용하세요. [[Feature #20205]]
+
+* 블록 인자를 가리키는 `it`이 추가됩니다. [[Feature #18980]]
+
+* 메서드 호출 시에 `nil`에 키워드 스플랫을 지원합니다.
+  `**nil`은 `**{}`와 비슷하게 동작하며, 키워드를 넘기지 않으며,
+  어떤 변환 메서드도 호출하지 않습니다. [[Bug #20064]]
+
+* 블록을 인덱스로 사용할 수 없게 됩니다. [[Bug #19918]]
+
+* 키워드 인수를 인덱스로 사용할 수 없게 됩니다. [[Bug #20218]]
+
+## YJIT
+
+TL;DR:
+* x86-64와 arm64 플랫폼에서 대부분의 벤치마크에서 성능이 향상되었습니다.
+* YJIT 메타 데이터 컴파일의 메모리 사용량이 줄었습니다.
+* 여러 버그가 수정되었습니다. YJIT는 이제 더 견고하고 잘 테스트되었습니다.
+
+새 기능:
+* YJIT의 메모리 사용량을 추적하는 통합 메모리 제한을 `--yjit-mem-size`
+  커맨드 라인 옵션(기본값 128 MiB)으로 추가했습니다.
+  이는 이전 `--yjit-exec-mem-size`보다 직관적이며, YJIT의 전체 메모리 사용량을 추적합니다.
```

Comment:
How about this?

```suggestion
* `--yjit-mem-size` 커맨드 라인 옵션(기본값 128 MiB)을 통해
  YJIT의 통합 메모리 제한을 추가했습니다.
  이는 이전 `--yjit-exec-mem-size`보다 직관적이며, YJIT의 전체 메모리 사용량을 추적합니다.
```

Reply:
Looks me better 👍

Reply:
fixed [1e55b9a](https://github.com/ruby/www.ruby-lang.org/pull/3438/commits/1e55b9aaed21d1e8cc4335b1f85cb75c81f674ca)